src/vm/threads.h

```diff
@@ -4838,8 +4838,8 @@ class Thread: public IUnknown
         LIMITED_METHOD_CONTRACT;
         PRECONDITION(HasPendingGCStressInstructionUpdate());
 
-        m_pbDestCode = NULL;
-        m_pbSrcCode = NULL;
+        VolatileStoreWithoutBarrier<BYTE*>(&m_pbDestCode, NULL);
+        VolatileStore<BYTE*>(&m_pbSrcCode, NULL);
```

Comment:
This guarantees the clear of dest occurs (is observed) before src, for similar reasons.